File: week4/507.py
for b in range(int(input())):
    s=int(input())
    N=[]
    for i in range(s-1):
        N.append(int(input()))
    Nsum=[0]*s
    for i in range(1,s):
        Nsum[i]=Nsum[i-1]+N[i-1]
    ans=-1000000
    for i in range(s-1):
        for j in range(i+1,s):
            if Nsum[j]-Nsum[i]>ans:
                ans=Nsum[j]-Nsum[i]
                ans_section=[i+1,j+1]
            elif Nsum[j]-Nsum[i]==ans and j-i>ans_section[1]-ans_section[0]:
                ans_section=[i+1,j+1]
    #print(Nsum)
    if ans<0:
        print("Route",b+1,"has no nice parts")
    else:
        print("The nicest part of route",b+1,"is between stops",ans_section[0],"and",ans_section[1])

File: week4/507.cpp
#include<bits/stdc++.h>
using namespace std;

int N[100000],Nsum[100000],ans_section[2],ans;

int main(){
    int B,s,b,i,j,sumi;
    Nsum[0]=0;
    cin>>b;
    for(B=1;B<=b;B++){
        cin>>s;
        for(i=0;i<s-1;i++)cin>>N[i];
        for(i=1;i<s;i++){
            Nsum[i]=Nsum[i-1]+N[i-1];
        }
        ans=-100000000;
        
        for(i=0;i<s-1;i++){
            for(j=i+1;j<s;j++){
                sumi=Nsum[j]-Nsum[i];
                if(sumi>ans){
                    ans=sumi;
                    ans_section[0]=i+1;
                    ans_section[1]=j+1;
                }else if(sumi==ans && j-i>ans_section[1]-ans_section[0]){
                    ans_section[0]=i+1;
                    ans_section[1]=j+1;
                }
            }
        }

        if(ans<0){
            cout<<"Route "<<B<<" has no nice parts\n";
        }else{
            cout<<"The nicest part of route "<<B<<" is between stops "<<ans_section[0]<<" and "<<ans_section[1]<<"\n";
        }
    }
}